divviup/src/main/java/org/divviup/android/Client.java

```diff
@@ -55,7 +62,76 @@ public static Client<Boolean> createPrio3Count(
             TaskId taskId,
             long timePrecisionSeconds
     ) {
-        return new Client<>(leaderEndpoint, helperEndpoint, taskId, timePrecisionSeconds, VDAF.PRIO3COUNT);
+        return new Client<>(
+                leaderEndpoint,
+                helperEndpoint,
+                taskId,
+                timePrecisionSeconds,
+                VDAF.PRIO3COUNT,
```

Comment:
design: currently, we store an enum determining the type, as well as `length`, `bits`, and `chunkLength` that parameterize some VDAFs (and are ignored in other VDAFs). This works, but having most VDAF clients hold a bunch of parameters that are unused is a bit confusing.

Separately, `sendMeasurement` has a giant if-block/switch over the set of supported VDAFs. This will get a bit unwieldy as we support additional VDAFs.

I think we could improve both issues by creating a (private-to-Client) `ReportPreparer` interface which has a `prepareReport` method which is effectively the "body" of the switch block that exists in `sendMeasurement` (i.e. it calls into the Rust code after performing some basic validation/conversions/copies). Implement ReportPreparer once per VDAF; have the constructor take the necessary `length`/`bits`/`chunkLength` parameters; and store a ReportPreparer in `Client` instead of `vdaf`/`length`/`bits`/`chunkLength`.